### PR TITLE
CIRC-1696 Fix issue with searching the least recalled loans

### PR DESCRIPTION
**Pull request by roman-barannyk**
## Purpose
- Fix issue with searching the least recalled loans;

Resolves: [CIRC-1696](https://issues.folio.org/browse/CIRC-1696)

## Approach
- Ignore loans with not recallable items;
- Add test;

### CODE REVIEW

File: src/test/java/api/requests/RequestsAPICreationTests.java [hunk 1]
```diff
@@ -4240,6 +4278,19 @@ private ItemResource buildItem(UUID instanceId, String barcode, IndividualResour
...
 
+  private ItemResource buildItem(UUID instanceId,
+    Function<ItemBuilder, ItemBuilder> additionalItemProperties) {
+
+    UUID isbnIdentifierId = identifierTypesFixture.isbn().getId();
+
+    return itemsFixture.basedUponSmallAngryPlanet(
+      holdingBuilder -> holdingBuilder.forInstance(instanceId),
+      instanceBuilder -> instanceBuilder
+        .addIdentifier(isbnIdentifierId, "9780866989732")
```
Comment (alexanderkurash) [inline src/test/java/api/requests/RequestsAPICreationTests.java:4289]: remove

File: src/test/java/api/requests/RequestsAPICreationTests.java [hunk 2]
```diff
@@ -3822,6 +3823,43 @@ void recallTlrShouldNotBeCreatedForInstanceWithOnlyAgedToLostItem() {
...
+    assertThat(firstRequestJson.getString("requestLevel"), is(RequestLevel.TITLE.getValue()));
+    assertThat(firstRequestJson.getString("requestType"), is(RECALL.getValue()));
+    assertThat(firstRequestJson.getString("instanceId"), is(instanceId.toString()));
+    assertThat(firstRequestJson.getString("itemId"), is(item.getId()));
+    assertThat(firstRequestJson.getJsonObject("item").getString("status"), is("Checked out"));
+    declareLostFixtures.declareItemLost(loan.getJson());
+
+    ItemResource onOrderItem = buildItem(instanceId, itemBuilder -> itemBuilder
+      .withBarcode("222")
+      .withMaterialType(materialTypesFixture.book().getId())
```
Comment (alexanderkurash) [inline src/test/java/api/requests/RequestsAPICreationTests.java:3848]: ?remove

File: src/test/java/api/requests/RequestsAPICreationTests.java [hunk 3]
```diff
@@ -3822,6 +3823,43 @@ void recallTlrShouldNotBeCreatedForInstanceWithOnlyAgedToLostItem() {
...
+
+    assertThat(firstRequest.getResponse(), hasStatus(HTTP_CREATED));
+    JsonObject firstRequestJson = firstRequest.getJson();
+    assertThat(firstRequestJson.getString("requestLevel"), is(RequestLevel.TITLE.getValue()));
+    assertThat(firstRequestJson.getString("requestType"), is(RECALL.getValue()));
+    assertThat(firstRequestJson.getString("instanceId"), is(instanceId.toString()));
+    assertThat(firstRequestJson.getString("itemId"), is(item.getId()));
+    assertThat(firstRequestJson.getJsonObject("item").getString("status"), is("Checked out"));
+    declareLostFixtures.declareItemLost(loan.getJson());
+
```
Comment (alexanderkurash) [inline src/test/java/api/requests/RequestsAPICreationTests.java:3845]: check that status changed to Declared lost

**Review by tatsiana-tarhonskaya — APPROVED** [review]

File: src/test/java/api/requests/RequestsAPICreationTests.java (hunk 1 above)
Comment (roman-barannyk) [reply src/test/java/api/requests/RequestsAPICreationTests.java]: done

File: src/test/java/api/requests/RequestsAPICreationTests.java (hunk 3 above)
Comment (roman-barannyk) [reply src/test/java/api/requests/RequestsAPICreationTests.java:3848]: done

File: src/test/java/api/requests/RequestsAPICreationTests.java (hunk 2 above)
Comment (roman-barannyk) [reply src/test/java/api/requests/RequestsAPICreationTests.java]: done

**Pull request merged**